### PR TITLE
Rest Framework Tests: Improve speed and repeatability

### PR DESCRIPTION
**Pull request by Maffooch**
- ViewSets: Support `get_queryset` function for more flexible queryset manipulation
- Authorization helpers: Specify order by id for most queries. For queries that already had a specified ordering, it was left alone if the field was unique on the model, such as Products. Otherwise, the ordering was changed to `id`
- Viewsets: enagegement checklist serializer was not used, so removed
- Tests: Accomodate the enforced ordering
- Tests: Reduce logs for CICD tests to speed up testing overall by eliminating write operations
- Tests: Enable FailFast mode ("Tells Django to stop running the test suite after first failed test.")
- Tests: Enable shuffle tests to identify areas of tests that are dependent on each other
- Tests: Enable parallel tests to drastically reduce testing time
  - In local testing, I went from ~240 seconds to ~45 seconds

[sc-6766]

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

**Comment by Maffooch** [comment]
@kiblik I was not planning on figuring out the `shuffle` issue quite yet, but addressing inconsistent ordering in API responses largely corrected those problems. I did not intended to steal your thunder, and I hope you do not feel that way 🙏🏼 

 I also did not plan on the `parallel` flag either, but I ran it just for kicks and encountered a different error this time around. It keyed me into to looking at how files were passed as class variables. Files, stack traces, and a few other things are not able to be serialized. Anyway, I thought you'd like to know what the underlying issue was 😄

File: unittests/test_jira_config_engagement.py [hunk 1]
```diff
@@ -265,9 +266,10 @@ def test_add_jira_project_to_engagement_without_jira_project(self, jira_mock):
     @patch('dojo.jira_link.views.jira_helper.is_jira_project_valid')
     def test_add_empty_jira_project_to_engagement_without_jira_project(self, jira_mock):
         jira_mock.return_value = True  # cannot set return_value in decorated AND have the mock into the method
-        engagement = self.add_engagement_without_jira_project(expected_delta_jira_project_db=0)
-        self.empty_jira_project_for_engagement(engagement, expected_delta_jira_project_db=0)
-        self.assertEqual(jira_mock.call_count, 0)
+        with contextlib.suppress(ValueError):
```
Comment (cneill) [inline unittests/test_jira_config_engagement.py:269]: Could you add a comment about why suppressing ValueErrors is necessary here?

File: unittests/test_jira_config_engagement.py [hunk 2]
```diff
@@ -279,15 +281,16 @@ def test_edit_jira_project_to_engagement_with_jira_project(self, jira_mock):
...
-        engagement = self.add_engagement_with_jira_project(expected_delta_jira_project_db=1)
-        # clearing out jira config used to be possible. what todo?
-        # - delete jira project? would disconnect all existing jira issues in defect dojo from the config?
-        # - allow jira project with empty jira instance and/or empty project_key? unpredictable behaviour
-        # - so prevent clearing out these values
-        # response = self.empty_jira_project_for_engagement(Engagement.objects.get(id=3), -1)
-        # expecting ValueError as we can't delete existing JIRA Projects
-        self.empty_jira_project_for_engagement(engagement, expected_delta_jira_project_db=0, expect_error=True)
-        self.assertEqual(jira_mock.call_count, 1)
+        with contextlib.suppress(ValueError):
```
Comment (cneill) [inline unittests/test_jira_config_engagement.py:284]: Same as above - could you add a quick comment explaining this line?

**Review by cneill — COMMENTED** [review]
Just a couple spots I wasn't sure about - otherwise this looks good! Very excited for faster, more stable test runs 😄

**Review by cneill — APPROVED** [review]

**Comment by kiblik** [comment]
> @kiblik I was not planning on figuring out the `shuffle` issue quite yet, but addressing inconsistent ordering in API responses largely corrected those problems. I did not intended to steal your thunder, and I hope you do not feel that way 🙏🏼
> 
> I also did not plan on the `parallel` flag either, but I ran it just for kicks and encountered a different error this time around. It keyed me into to looking at how files were passed as class variables. Files, stack traces, and a few other things are not able to be serialized. Anyway, I thought you'd like to know what the underlying issue was 😄

No hard feelings 😄 I will be happy when it will be solved

**Review by hblankenship — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**